src/main/java/com/minecolonies/coremod/colony/buildings/BuildingSmeltery.java

```diff
@@ -0,0 +1,198 @@
+     */
+    private static final String TAG_FURNACES = "furnaces";
+
+    /**
+     * List of registered furnaces.
+     */
+    private final List<BlockPos> furnaces = new ArrayList<>();
+
+    /**
+     * Instantiates a new cook building.
+     *
+     * @param c the colony.
+     * @param l the location
+     */
+    public BuildingSmeltery(final Colony c, final BlockPos l)
+    {
+        super(c, l);
+    }
+
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return SMELTERY_DESC;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return MAX_BUILDING_LEVEL;
+    }
+
+    @NotNull
+    @Override
+    public AbstractJob createJob(final CitizenData citizen)
+    {
+        return new JobSmelter(citizen);
+    }
+
+    @NotNull
+    @Override
+    public String getJobName()
+    {
+        return SMELTERY_DESC;
+    }
+
+    /**
+     * Return a list of furnaces assigned to this hut.
+     *
+     * @return copy of the list
+     */
+    public List<BlockPos> getFurnaces()
+    {
+        return new ArrayList<>(furnaces);
+    }
+
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.writeToNBT(compound);
+        @NotNull final NBTTagList furnacesTagList = new NBTTagList();
+        for (@NotNull final BlockPos entry : furnaces)
+        {
+            @NotNull final NBTTagCompound furnaceCompound = new NBTTagCompound();
+            furnaceCompound.setTag(TAG_POS, NBTUtil.createPosTag(entry));
+            furnacesTagList.appendTag(furnaceCompound);
+        }
+        compound.setTag(TAG_FURNACES, furnacesTagList);
+    }
+
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        final NBTTagList furnaceTagList = compound.getTagList(TAG_FURNACES, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < furnaceTagList.tagCount(); ++i)
+        {
+            furnaces.add(NBTUtil.getPosFromTag(furnaceTagList.getCompoundTagAt(i).getCompoundTag(TAG_POS)));
+        }
+    }
+
+    @Override
+    public void registerBlockPosition(@NotNull final Block block, @NotNull final BlockPos pos, @NotNull final World world)
+    {
+        super.registerBlockPosition(block, pos, world);
+        if (block instanceof BlockFurnace && !furnaces.contains(pos))
+        {
+            furnaces.add(pos);
+        }
+        markDirty();
+    }
+
+    @SuppressWarnings(MAGIC_NUMBERS_SHOULD_NOT_BE_USED)
+    public int ingotMultiplier(final int citizenLevel, final Random random)
+    {
+        switch(getBuildingLevel())
+        {
+            case 1:
+                return random.nextInt(ONE_HUNDRED_PERCENT - citizenLevel) == 0 ? DOUBLE : 1;
+            case 2:
+                return random.nextInt(ONE_HUNDRED_PERCENT - citizenLevel * DOUBLE) == DOUBLE ? 2 : 1;
+            case 3:
+                return 2;
+            case 4:
+            case 5:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)